src/components/nav-icon-item/styles.ts

```diff
@@ -0,0 +1,61 @@
+import { styled } from '@linaria/react'
+import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react'
+
+type CommonNavIconItemProps = {
+  /**
+   * NOTE: data attribute to indicate the state of the nav item
+   * by default the styles is already applied with pseudo-class selector
+   * but to make it more flexible, consumer could use this attribute to
+   * change the styles
+   */
+  'data-state'?: 'focus' | 'hover' | 'active'
+}
+
+const baseStyles = `
+  display: inline-flex;
+  padding: var(--spacing-2, 8px);
+  justify-content: center;
+  align-items: center;
+  gap: var(--spacing-none, 0px);
+  border-radius: var(--corner-default, 0.25rem);
+  background: var(--white, #ffffff);
+  border: none;
+  outline: none;
+
+  /* change children color by default */
+  > * {
+    color: var(--neutral-400, #798da1) !important;
+  }
+
+  &:focus, &[data-state='focus'] {
```

Comment:
**suggestion:** I _think_ (not 100% sure) we would prefer `:focus-visible`, instead of `:focus`, as the focus outline really only needs to appear when focus is applied via keyboard interaction, not pointers.

Reply:
> I think (not 100% sure) we would prefer :focus-visible, instead of :focus, as the focus outline really only needs to appear when focus is applied via keyboard interaction, not pointers

My reason for using `:focus` instead of `:focus-visible` is to support the most browser compatibility, especially for Safari users (it’s introduced in 15.4); but I think we could set a fallback for it.

Reply:
Ah, that's interesting. Do we know what devices that relates to? I only ask because `:focus-visible` is part of the Widely Available Baseline, which (personally) I think is reasonable for us to use as our minimum standard. I've not had to deal much with mobile browser compatibility in the past though, so more than happy to take the lead on this stuff from you and @adrian-reapit given it's directly applicable to the Reapit Go product 👍  I think `:focus` is just going to be a bit "in your face" for most desktop users 🤔